src/software/SfM/main_SfMInit_ImageListingFromKnownPoses.cpp

```diff
@@ -0,0 +1,266 @@
+
+
+  // Configure an empty scene with Views and their corresponding cameras
+  SfM_Data sfm_data;
+  sfm_data.s_root_path = sImageDir; // Setup main image root_path
+  Views & views = sfm_data.views;
+  Poses & poses = sfm_data.poses;
+  Intrinsics & intrinsics = sfm_data.intrinsics;
+  
+
+  C_Progress_display my_progress_bar( vec_image.size(),
+      std::cout, "\n- Image listing From Known Poses- \n" );
+  std::ostringstream error_report_stream;
+  for ( std::vector<std::string>::const_iterator iter_image = vec_image.begin();
+    iter_image != vec_image.end();
+    ++iter_image, ++my_progress_bar )
+  {
+   
+    const std::string sImageFilename = stlplus::create_filespec( sImageDir, *iter_image );
+    const std::string sImFilenamePart = stlplus::filename_part(sImageFilename);
+
+    // Test if the image format is supported
+    if (openMVG::image::GetFormat(sImageFilename.c_str()) == openMVG::image::Unknown)
+    {
+      error_report_stream
+          << sImFilenamePart << ": Unkown image file format." << "\n";
+      continue; // image cannot be opened
+    }
+
+
+    if (sImFilenamePart.find("mask.png") != std::string::npos
+       || sImFilenamePart.find("_mask.png") != std::string::npos)
+    {
+      error_report_stream
+          << sImFilenamePart << " is a mask image" << "\n";
+      continue;
+    }
+
+
+    // Test if this image can be read
+    ImageHeader imgHeader;
+    if (!openMVG::image::ReadImageHeader(sImageFilename.c_str(), &imgHeader))
+      continue; // image cannot be read
+
+ 
+    //read ground truth from file
+    // Test if ground truth file exists
+    std::string sGTFilePath = stlplus::create_filespec(sGroundTruthDir,sImFilenamePart+suffix);
+    if (!stlplus::is_file( sGTFilePath ) )
+    {
+      error_report_stream << std::endl << "There is not valid GT data to read from " << sGTFilePath << std::endl;
+      continue;
+    }
+    else
+    {
+      // Load GT
+      cameras::PinholeCamera cam;
+      double width = 0,height = 0;
+      if ( read_Strecha_Camera(sGTFilePath, cam, width, height) )
+      {
+        Mat3 K = cam._K;
+        double focal = (K(0,0)+K(1,1))/2.0;//it will be better K(0,0)==K(1,1)
+        double pxx   = K(0,2);
+        double pyy   = K(1,2);
+
+        std::shared_ptr<View> view                   = std::make_shared<View>(*iter_image, views.size(), views.size(), views.size(), width, height);
+        Pose3 pose                                   = Pose3(cam._R,cam._C);
+        std::shared_ptr<Pinhole_Intrinsic> intrinsic = std::make_shared<Pinhole_Intrinsic>(width,height,focal,pxx,pyy);
+
+        // Add the view to the sfm_container
+        views[view->id_view] = view;
+        // Add the pose to the sfm_container
+        poses[view->id_pose] = pose;
+        // Add the intrinsic to the sfm_container
+        intrinsics[view->id_intrinsic] = intrinsic;
+
+      }
+      else
+      {
+        continue;
+      }
+    }
+    
+  }
+
```

Comment:
We can add an option to try to merge the intrinsic that have exactly the same value.
See here: https://github.com/openMVG/openMVG/blob/master/src/software/SfM/main_SfMInit_ImageListing.cpp#L465

Reply:
Actually I know there is a b_Group_camera_model which can merge the intrinsic. 
The reason I didn't use it is that I could not make sure that all the views from the ground truth data have the same intrinsic.

Reply:
The function will merge the intrinsic only if the values of the parameter are exactly the same (Hash of the parameters values).

Reply:
Thanks I will try this and update my code.